content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -1148,17 +1228,86 @@ public void webhookEvent(HttpServletRequest request, String providerName, Option
 			}
 		} catch (Exception e) {
 			log.error(e.getMessage(), e);
+			errors++;
 		} finally {
 			if (bufferedReader != null) {
 				try {
 					bufferedReader.close();
 				} catch (Exception e) {
 					log.error(e.getMessage(), e);
+					errors++;
 				}
 			}
 		}
+
 		body = stringBuilder.toString();
-		log.info(body);
+		JSONObject webhookJSON = JSONObject.fromObject(body);
+		String eventType = request.getHeader("X-Turnitin-Eventtype");
+		String signature_header = request.getHeader("X-Turnitin-Signature");
+
+		// Make sure cb is signed correctly
+		boolean callback_correctly_signed = false;
+		try {
+			String secrete_key_encoded = getSigningSignature(apiKey.getBytes(), body);
+			if (secrete_key_encoded != null && signature_header.equals(secrete_key_encoded)) {
+				callback_correctly_signed = true;
+			}
+		} catch (Exception e1) {
+			log.error(e1.getMessage(), e1);
+		}
+
+		if (callback_correctly_signed) {
+			try {
+				if (eventType.equals("SUBMISSION_COMPLETE")) {
```

Comment:
make "SUBMISSION_COMPLETE" a constant and switch the "equals" logic to "SUBMISSION_COMPLETE".equals(eventType)